1. Listázzuk a <span style="color: #800000;">`kolcsonzes`</span> tábla következő oszlopait: ID, ki\_datum, vissza\_datum! 

  

a.  Az ID oszlop neve legyen Azonosító

b.  A ki\_datum oszlop neve legyen Kölcsönzés\_dátuma

c.  A vissza\_datum oszlop neve legyen Visszahozatal\_dátuma

In [ ]:
SELECT ID AS 'Azonosító', -- a kölcsönző azonosítója
        ki_datum AS 'Kölcsönzés dátuma',
        vissza_datum AS 'visszahozatal dátuma'
From kolcsonzesek

2. Melyek azok a kölcsönzések, amelyek 2005.01.01 és 2007.01.01 között történtek?

In [ ]:
SELECT  *
FROM kolcsonzesek
WHERE ki_datum > '2005.01.01' and ki_datum < '2007.01.01'
-- WHERE li_datum BETWEEN '2005.01.01' AND '2007.01.01'

3. Melyek azok a tagok, akiknek kedvenc stílusa a játék, és nevük a-ra végződik?

  

a.  Csak a tagok neve jelenjen meg, az oszlop neve Név legyen

In [ ]:
SELECT nev AS Név
from tagok
WHERE nev LIKE '%a' and kedvencstilus = 'játék'

4. Listázzuk egy oszlopban a tagok nevét, címét és kedvenc stílusát összefűzve! 

  

a.  A nevet, címet és a stílust kötőjellel válasszuk el

b.  A kötőjelek előtt és után legyen 1-1 szóköz

c.  Az oszlop neve legyen TAG

d.  Kinga keresztnevű tagok ne legyenek benne a listában

In [ ]:
select nev+' - '+cim+' - '+kedvencstilus AS TAG
from tagok
where nev NOt LIKE '%Kinga%'

5. Mennyi lenne a dvd-k nettó ára, ha 20%-kal csökkenne? 

  

a.  Csak a dvd-k címét és a csökkentett árat jelenítsük meg

b.  A csökkentett árat nevezzük el diszkont\_ar-nak

In [ ]:
-- diszkont ár = nettó ár * 0,8
select cim as 'Cím',
nettoar * 0.8 AS 'diszkont_ar'
from dvd

6. Melyek azok a kölcsönzések, ahol még nem hozták vissza a dvd-t?

In [ ]:
select *
from kolcsonzesek
where vissza_datum is null

7. Listázzuk azon diákok nevét és születési idejét, akik 1984-ben vagy 1985-ben születtek!  

  

a.  A születési dátum csak az évet, hónapot és a napot tartalmazza! 

b.  A születési dátum oszlopát nevezzük el 'Születési idő'-nek!

In [ ]:
select Nev,
        -- cast(Szulido as date) as 'Születési idő'
        -- left(cast(Szulido AS nvarchar(25)), 11)
        cast(YEAR(Szulido) as nvarchar(4)) + '.' +
        cast(month(Szulido) as nvarchar(2)) + '.' +
        cast(day(Szulido) as nvarchar(2))
from diak
where YEAR(Szulido) = 1984 or YEAR(Szulido) = 1985
-- WHERE YEAR(Szulido) IN (1984, 1985)

8. Készítsünk lekérdezést, amely a tanulók nevét és az ebből képzett felhasználói nevét tartalmazza. 

  

a.  A felhasználói név álljon a név első kettő, illetve utolsó kettő betűjének összefűzéséből!

b.  A listát szűrjük azon tanulókra, akik teljes neve - szóközzel együtt - legalább 10 betűs!

In [68]:
-- LEFT, RIGHT, LEN
select Nev,
        left(Nev,2) + right(Nev,2) as 'Felhasználói név'
from diak
where len(Nev) >= 10

(8 rows affected)

Total execution time: 00:00:00.008

Nev Felhasználói név Kovács Imre Kore Horváth Emil Hoil Kapos Petra Kara Csóka Anna Csna Nyúl Tamás Nyás Ordasi Emma Orma Koppány Olga Koga Kozma Patrícia Koia

9. Készítsünk listát a diákok adatairól, ahol a diákok neve úgy jelenik meg, hogy először a keresztnév, majd a vezetéknév, közöttük szóközzel!  

a.  Ezen felül a születési idő három külön oszlopban jelenjen meg (év, hónap, nap). 

b.  Az oszlopokat nevezzük el értelemszerűen!

10. Készítsünk listát a munkák azonosítójáról és megnevezéséről! Egy új oszlopban azt is jelenítsük meg, hogy melyik munka hetente mennyit fizet! 

a.  Az oszlop neve legyen Heti bér, az értéket kerekítsük 1000 forintra! 

b.  A listát szűrjük azon rekordokra, ahol a kerekített heti bér 10000 Ft felett van!

In [69]:
-- heti bér = óradíj * óraszám * 5
-- ROUND fv
select munkaId,
        allas, 
        ROUND(oradij * oraszam * 5, -3) as 'Heti bér'
from munka
where  ROUND(oradij * oraszam * 5, -3) > 10000

(7 rows affected)

Total execution time: 00:00:00.002

munkaId allas Heti bér 5 kézbesítő 14000 7 eladó 12000 16 pincér 14000 17 pincér 14000 20 takarító 11000 21 takarító 11000 41 eladó 12000

11. A diákok számára differenciált béremelést terveznek: a középiskolások esetében 33%, egyéb esetben 17% mértékben. Készítsünk listát, amely tartalmazza a munkák azonosítóját, az állás nevét, az eredeti óradíjat és a tervezett emelt óradíjat.

  

a.  Az oszlopoknak adjuk nevet értelemszerűen!

b.  Az emelt óradíj összegét kerekítsük egészre! (Az esetlegesen megjelenő 0 tizedesjegyekkel ne foglalkozzunk)

12. A diákok számára próbaidőt írnak elő, amely a munka kezdetétől számítva 3 hónapig tart.    Jelenítsük meg a munka tábla adatait egy új oszloppal kiegészítve, amelyik a próbaidő    végének dátumát mutatja!  

  

a.  Az oszlop neve Próbaidő vége legyen! 

b.  A lista ne tartalmazza azokat a munkákat, ahol a diákok azonosítója nincs megadva!

13. Készítsünk listát a munkák azonosítóiról és a megnevezésükről! A megnevezés allas-óraszám formában jelenjen meg, pl: eladó-4. 

  

a.  Jelenítsük meg a dátumot is, de csak a évet és a hónapot, pl: 2003-07! 

b.  Ennek az oszlopnak 'Kezdés hónapja' legyen a neve! A listát szűrjük a nyári hónapokra!

14. Készítsünk listát a diákok adatairól, amely a Nev oszlop helyén a diák monogramját jeleníti meg (keresztnév, illetve vezetéknév első betűje összefűzve). 

a.  A listát szűrjük 1983.01.01 és 1987.06.01 között született diákokra!

15. A diákok a munkakezdés évének utolsó napján bónuszt kapnak, amennyiben az adott évben legalább 6 hónapot dolgoznak. 

  

a.  Jelenítsük meg, hogy az egyes munkák adatait, egy új oszlopban az éves bónusz napját is 'Bónusz dátum' néven! 

b.  A lista csak a bónuszra jogosultak adatait tartalmazza!